weblate/formats/base.py

```diff
@@ -276,7 +276,7 @@ def has_unit(self) -> bool:
         return self.unit is not None
 
     def clone_template(self) -> None:
-        self.mainunit = self.unit = copy(self.template)
+        self.mainunit = self.unit = deepcopy(self.template)
```

Comment:
We used to have this, but this scales horribly: when adding 100 units, you need to clone 100 times the underlying objects. See 8df491b98ed372aaddcc611234d0dd486737ef51

Reply:
I see :-/ I'm concerned that this can actually lead to bugs.

My plan:
- Extract PHP test fix (IMO worth merging on its own): #11999
- Figure out of the shallow copy causes end-user visible issues, then take it from there.

Reply:
Yes, it can lead to problems (I think we had this in the past). I think the only problematic piece is `xmlelement` attribute for LISA-based units, what can be addressed in several ways. Maybe doing a copy of it before wiping the notes would be enough to address this issue?

Reply:
Hmmm... I'm a bit concerned that this might lead to a lot of special casing down the line.

I'm wondering if rather than copying the whole unit, creating a new unit and copying only what we care about could be a viable approach. Have you ever toyed with this idea? (I have not the slightest idea what the implications for various formats would be).

FWIW: I have not managed to find any (trivial) issues caused by a shallow copy so far (I have found #12000, but you already fixed that in record time :P).

Reply:
I think shallow copy could cause issue in case both template and translation store are being updated. Depending on the ordering, it might wipe notes from the template. See #2144 / 0fa7a7ecf4eef66867be5bebf81b457738c7861c for what kind of issues this can cause.

Rebuilding new unit would be an option as well, but I'm afraid that the translate-toolkit API is very inconsistent in this regard. 

Looking at it, there is actually a `TranslationUnit.buildfromunit` method which might do what we need here. The question is whether it works reliably. Some formats even implement `copy` method where they deal with the xml, see https://github.com/translate/translate/blob/ae6abd4c8935c0675933eea49eaed7255dca7981/translate/storage/tmx.py#L105-L115. Polishing these in the translate-toolkit and using it from Weblate might be the cleanest approach.

Reply:
- I have opened a CI only PR (#12015) to check what happens if we use `buildFromUnit` in `clone_template`.
- It seems like the deepcopy issue we see in the unit tests does not actually affect usage (I've tested this in a local dev instance). I will try to adjust the test cases to roundtrip via actual files (which is probably closer to actual usage anyways).

Reply:
deepcopy is problematic as it can copy huge data in the background several times. Imagine 50 MB XLIFF file where you add 20 strings. You would suddenly need 1 GB of memory just for the deep copies of the XML (you would actually need at least 2x more as that also copies the store with parsed units).